### PR TITLE
DEP: create new package.json and composer. json with approved packages.

### PR DESCRIPTION
**Pull request by lukastocker**
The list about the approved dependency packages can be found here: [JourFixe-2024-01-22](https://docu.ilias.de/ilias.php?baseClass=ilwikihandlergui&cmdNode=16n:r6&cmdClass=ilobjwikigui&cmd=viewPage&ref_id=1357&page=JourFixe-2024-01-22#ilPageTocA2513)

As mentioned in [PR 6653](https://github.com/ILIAS-eLearning/ILIAS/pull/6653#issuecomment-1964466578) I have not included the simplesamlphp package as it's currently not working with the Symfony Console package version 7. 

As you can see a commit with a new phpunit.xml file was also created, as it would not have worked otherwise. Please also see the change log of PHPUnit 10+, as many functions are deprecated: 
- [Change log](https://github.com/sebastianbergmann/phpunit/blob/10.5/ChangeLog-10.5.md) 
- [Deprecrations](https://github.com/sebastianbergmann/phpunit/blob/10.5/DEPRECATIONS.md)

### CODE REVIEW

File: components/ILIAS/Chatroom/chat/package.json [hunk 1]
```diff
@@ -3,10 +3,9 @@
   "description": "ILIAS CHAT",
   "version": "2.0.0",
   "dependencies": {
-    "async": "^3.2",
+    "async": "3.2",
```
Comment (mjansenDatabay) [inline components/ILIAS/Chatroom/chat/package.json:6]: As decided by the JF, we should use v. `3.x` for `winston` (a few lines below this change here), see: https://github.com/ILIAS-eLearning/ILIAS/pull/6769#issuecomment-1850061501.

We already tested this successfully (based on 9.x and an upgraded library and chat component) with v. `^3.13.0`

**Comment by lukastocker** [comment]
Hi @kergomard ,

I did some required adjustments.

**Newly included into composer:** 

- league/flysystem
- sabre/dav
- dflydev/fig-cookies
- symfony/console
- simplesamlphp

**Changed:**

- winston to ^3.x (NPM/Chatroom)

The issue regarding symfony/console and simplesamlphp was resolved, as you can see in the comments of following PR:
https://github.com/ILIAS-eLearning/ILIAS/pull/6653

**Comment by kergomard** [comment]
Thank you very much @lukastocker and @klees for all the effort you put into this!

There is still a long list of missing js-dependencies, but:
- `npm clean-install` + `composer install` + `php cli/setup.php update -y` run through and lead to a system one can log into and that is generally usable.
- `scripts/PHPUnit/run_tests.sh` can be run. It leads to a lot of deprecation warnings, but it runs.
- `npm test` cannot be run, as we need to move from mocha to jest, but the tests are currently commented out.

I will thus merge this now and we go on from there.

Thanks again and best,
@kergomard in the name of the Technical Board

**Comment by klees** [comment]
Dear colleagues,

this PR replaces the dependencies of ILIAS 9 by the dependencies that have been approved by the JF. If we'd missed anything we are sorry, just a lot of changes to incorporate here. Please add missing stuff via PR. There are some open ends that need to be tackled by individual contributors:

* [This commit](https://github.com/ILIAS-eLearning/ILIAS/pull/7141/commits/7aac5c91c665a77601b5fa2fa36a262a73e39b5a) uncomments files that have been pulled from some node-module that hasn't been requested until now. I will take this list to the JF to make it public there, but everybody might want to have a look into this already and tackle the problem for their component. There will be stuff that currently does not work as expected because of missing npm-packages.
* [This commit](https://github.com/ILIAS-eLearning/ILIAS/pull/7141/commits/f07026261d5cb65d2d1ef5143f173bf2e5e53077) introduces many changes to make tests PHPUnit 10 compatible as discussed [here](https://github.com/ILIAS-eLearning/ILIAS/pull/7427). It feels as if we have unearthed various tests that haven't been run before. We tried to fix them, but deliberately added `markTestSkipped` if we felt that there was more than a compatibility problem in some failing test. Please everyone, look into skipped tests in your component and decide what to do with them.
* [This commit here](https://github.com/ILIAS-eLearning/ILIAS/pull/7141/commits/f32923df22eede698fa47b09522e64832460abe9) contains tests that errored or failed due to missing PHP dependencies. This mostly concerns the test-library `Mockery`. Since this makes test fail very early I renamed files so PHPUnit does not pick these tests up anymore. Please look into the according tests. Remove the `TBD` from the filename to make PHPUnit run the test and see the error. From my POV we should try to get rid of `Mockery` completely.
* [This commit](https://github.com/ILIAS-eLearning/ILIAS/pull/7141/commits/3aa813091b14da78633a190037b4cc2f674d003d) deactivates js-tests in our CI pipeline. We decided to [move tests to jest](https://github.com/ILIAS-eLearning/ILIAS/pull/6945) and I currently try to figure out how we would tackle that problem.
* As @kergomard points out, the PHPUnit tests display deprecation warnings. Please look into these, as these will bite us sometimes later.
* There is some hidden dependency in tests of the T&A that makes the tests fail for some execution orders. 1718723968 is a seed that exhibits that behaviour. If you run tests and a lot of tests fail in the T&A: just run them again. You should hit some working seed soon.

That's it for the moment. I'll bring this up on the JF again, but I definitly propose to merge this now. We can install and log into the system. There will be open issues due to missing javascript-libs, but if we do not move on with the freshly approved dependencies we will run into problems very late in our dev-process.

Kind regards!

**Pull request merged**